stable/traefik/README.md

```diff
@@ -104,6 +104,8 @@ The following tables lists the configurable parameters of the Traefik chart and
 | `ssl.defaultCert`               | Base64 encoded default certficate                                    | A self-signed certificate                 |
 | `ssl.defaultKey`                | Base64 encoded private key for the certificate above                 | The private key for the certificate above |
 | `acme.enabled`                  | Whether to use Let's Encrypt to obtain certificates                  | `false`                                   |
+| `acme.dnsProvider.name`         | Which DNS provider to use. See [here](https://github.com/xenolf/lego/tree/master/providers/dns) for the list of possible values. | `nil`                                     |
+| `acme.dnsProvider.$name`        | The configuration environment variables (encoded as a secret) needed for the DNS provider to do DNS challenge. See [here](#example-aws-route-53). | `{}`                     |
```

Comment:
If only one provider can be configured at a time, do we need to support each of these with separate yaml keys or could it be flattened to something like `acme.dnsProvider.envConfig`? I don't have a specific concern, just think it might be more complex than required.

Reply:
@c-knowles 

> If only one provider can be configured at a time, do we need to support each of these with separate yaml keys or could it be flattened to something like acme.dnsProvider.envConfig

That's similar to how I [originally implemented](https://github.com/kubernetes/charts/pull/1299/commits/70386896e22963e189da195191f05fb4f2772359), but it's difficult to tie down the schema if we allow a blank `envConfig`. As @krancour pointed out, we want it to be clear to the user what keys are expected here instead of having them to consult the documentation of lego.

Another added benefit here is you can have multiple cloud DNS provider configured here but you may want to switch between them. This implementation allows the user to only change the `dnsProvider.name` field without touching anything else.

Reply:
Ok sure, makes sense. I had missed the original changes. I think the keys could still be documented the other way as well but if the intention is that there’d eventually be a schema for the config then it’ll be easier like this with separate keys.